Source/Result.swift

```diff
@@ -34,6 +34,29 @@ public enum Result<Value> {
     case Success(Value)
     case Failure(NSData?, NSError)
 
+    /**
+        Creates a `.Success` result with the given value.
+
+        - parameter value: The value to store in the `.Success`.
+
+        - returns: A new `.Success` result.
+    */
+    public init(value: Value) {
```

Comment:
That's really a good point. I was a bit torn as well. After adding the tests, I realized it's not any easier to use the initializers than the default case initialization. Maybe this whole PR should just be declined. What you think @kylef?